packages/api/internal/handlers/sandbox_logs.go

```diff
@@ -0,0 +1,94 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/grafana/loki/pkg/loghttp"
+	"github.com/grafana/loki/pkg/logproto"
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/e2b-dev/infra/packages/api/internal/api"
+	"github.com/e2b-dev/infra/packages/api/internal/constants"
+	"github.com/e2b-dev/infra/packages/api/internal/utils"
+
+	"github.com/e2b-dev/infra/packages/shared/pkg/models"
+	"github.com/e2b-dev/infra/packages/shared/pkg/telemetry"
+)
+
+const (
+	oldestLogsLimit = 168 * time.Hour // 7 days
+)
+
+func (a *APIStore) GetSandboxesSandboxIDLogs(
+	c *gin.Context,
+	sandboxID string,
+	params api.GetSandboxesSandboxIDLogsParams,
+) {
+	ctx := c.Request.Context()
+	sandboxID = utils.ShortID(sandboxID)
```

Comment:
~shouldn't we check if it belongs to the team? But I wouldn't do it in this PR~

it's handled by team.id in query ✔️